Cython/Compiler/PyrexTypes.py

```diff
@@ -1649,11 +1649,10 @@ def declaration_code(self, entity_code,
                 entity_code = "*%s" % entity_code
         return self.base_declaration_code(base_code, entity_code)
 
-    def type_test_code(self, py_arg, notnone=False):
-
+    def type_test_code(self, code, py_arg, notnone=False):
         none_check = "((%s) == Py_None)" % py_arg
         type_check = "likely(__Pyx_TypeTest(%s, %s))" % (
-            py_arg, self.typeptr_cname)
+            py_arg, code.name_in_module_state(self.typeptr_cname))
```

Comment:
I think the issue here is that we'll eventually end up with different ways of looking up the module state depending on the type of the scope. I'll look into if I can improve this though

Reply:
I'm struggling to refactor this into something better.

Tangentially https://github.com/cython/cython/issues/6320 is another case where having `code` in this function could be useful - the string this function generates requires utility code that it can't include itself.